conn.go

```diff
@@ -339,6 +339,15 @@ func DialOpen(d Dialer, name string) (_ driver.Conn, err error) {
 	if err != nil {
 		return nil, err
 	}
+
+	// cn.ssl and cn.startup panic on error. Make sure we don't leak cn.c.
+	panicking := true
+	defer func() {
+		if panicking {
+			cn.c.Close()
+		}
+	}()
+
 	cn.ssl(o)
```

Comment:
It would be ideal if these functions returned errors instead of panicking... but this works as a hot fix. Would leave that to maintainers.

https://github.com/golang/go/wiki/CodeReviewComments#dont-panic

Reply:
Indeed. Ideally `cn.startup` would return an error. But that's a larger change, so I'm fine with this now.